pkg/cache/task.go

```diff
@@ -415,3 +416,21 @@ func (task *Task) enterState(event *fsm.Event) {
 		zap.String("destination", event.Dst),
 		zap.String("event", event.Event))
 }
+
+func PublishTaskEvent(taskID, reason, msg string, app interfaces.ManagedApp) error {
+	log.Logger.Info("taskID", zap.String("taskID", taskID))
+	task, err := app.GetTask(taskID)
+	if err != nil {
+		return fmt.Errorf("could not find %s task belonging to %s app", taskID, app.GetApplicationID())
+	}
+	pod := task.GetTaskPod()
+	if pod == nil {
+		return fmt.Errorf("could not obtain %s task's pod", taskID)
+	}
+
+	// TODO remove this
+	log.Logger.Debug("Emitting event", zap.String("pod name", pod.ObjectMeta.Name), zap.String("reason", reason), zap.String("message", msg))
+	events.GetRecorder().Event(pod, v1.EventTypeWarning, reason, msg)
+
+	return nil
```

Comment:
Do we need to return here explicitly?

Reply:
We have to return an `error` object, and in the good case it's a `nil` object. So I think, we should return here indeed.